app/controllers/votes_controller.rb

```diff
@@ -0,0 +1,33 @@
+class VotesController < ApplicationController
+  before_action :find_micropost
+  
+  def create
+
+    if already_voted?
+      flash[:warning] = "You can't vote more than once"
+    else
+      # I know it is not safely because someone can hack database by coding in form (or somehow).
+      # Does validation in model (1..5) solve this problem ?
+     @vote = @micropost.votes.create(grades: params[:grades], user_id: current_user.id) 
+      if @vote.save
+        redirect_to root_url
+      else
+        flash[:warning] = "1, 2, 3, 4 or 5 only"
```

Comment:
We could have an error that's not related to scores, so why would we hardcode error message that way?

Rails has inbuilt ways of handling errors and errors messages: https://guides.rubyonrails.org/active_record_validations.html#displaying-validation-errors-in-views